server/src/main/java/com/cloud/network/Ipv6ServiceImpl.java

```diff
@@ -167,19 +168,55 @@ private boolean isPublicIpv6PlaceholderNic(NicVO nic) {
         return new Pair<>(nic.getIPv6Address(), vlanOptional.get());
     }
 
+    private void publishPublicIpv6AssignActionEvent(final Network network, final String ipv6Address) {
+        String event = EventTypes.EVENT_NET_IP6_ASSIGN;
+        final String description = String.format("Assigned public IPv6 address: %1$s for network ID: %2$s. Subnet: %3$s, gateway: %1$s", ipv6Address,  network.getUuid(), network.getIp6Cidr());
+        ActionEventUtils.onCompletedActionEvent(CallContext.current().getCallingUserId(), network.getAccountId(), EventVO.LEVEL_INFO, event, description, network.getId(), ApiCommandResourceType.Network.toString(), 0);
+        final boolean usageHidden = networkDetailsDao.isNetworkUsageHidden(network.getId());
+        final String guestType = Vlan.VlanType.VirtualNetwork.toString();
+        UsageEventUtils.publishUsageEvent(event, network.getAccountId(), network.getDataCenterId(), 0L,
+                ipv6Address, false, guestType, false, usageHidden,
+                IPv6Network.class.getName(), null);
+    }
+
+    private void publishPublicIpv6ReleaseActionEvent(Network network, String nicIpv6Address) {
+        String event = EventTypes.EVENT_NET_IP6_RELEASE;
+        String description = String.format("Releasing public IPv6 address: %1$s from network ID: %2$s. Subnet: %3$s, gateway: %1$s", nicIpv6Address,  network.getUuid(), network.getIp6Cidr());
```

Comment:
@shwstppr 
since the operation is completed, it might be better to use `Released` not `Releasing` in the event description

```
         uuid: a4f141e4-a297-4fe4-9739-78840e720997
         type: NET.IP6RELEASE
        state: Completed
  description: Releasing public IPv6 address: fd00:10:11:51:1c00:88ff:fe00:3 from network ID: 698ae539-fb4d-4862-b69d-edd39d4460e1. Subnet: 2022:2:2:9::/64, gateway: fd00:10:11:51:1c00:88ff:fe00:3
```

Reply:
@weizhouapache done